### DISCUSSION TITLE
Error in watchdogd when generic monitor script is enabled.

### DISCUSSION THREAD

**Discussion by senthilnthangaraj** [troglobit/watchdogd#38] [Q&A] [answered]
I've encountered an issue with the watchdogd service when attempting to activate a generic monitoring script. This particular script has a runtime exceeding one second, leading to the system ultimately rebooting.

Our configuration is as follows:

generic {
   enabled = true
   interval = 60
   timeout = 20
   warning  = 1
   critical = 10
   monitor-script = "/usr/sbin/my-script.sh"
}

The error message we're encountering reads:

"Monitor script PID 1057 is still running after 20 seconds.
watchdog: watchdog0: watchdog did not stop!"

Upon further investigation, I've identified that the problem lies in the fact that 'gs->script_runtime' is measured in milliseconds, while 'gs->script_runtime_max' is maintainted in seconds, as indicated by the source code here: [link to source code](https://github.com/troglobit/watchdogd/blob/master/src/generic.c#L68).

This appears to be a bug that requires rectification. Kindly confirm if my assessment is accurate.

**Comment by troglobit** [comment]
Nice catch! I've created issue #39 for this, contributions to fix this (PR, or patches) are most welcome since I'm quite busy with other projects during the next couple of months.

**Comment by senthilnthangaraj** [comment]
Thank you ! 

I have created a PR for the same.

https://github.com/troglobit/watchdogd/pull/40

Can you please review !

**Reply by senthilnthangaraj** [reply, marked as answer]
Thanks @troglobit for merging this change https://github.com/troglobit/watchdogd/pull/40, closing the discussion.
